File: apps/api-gql/internal/delivery/gql/mappers/overlays-be-right-back.go
package mappers

import (
	"github.com/twirapp/twir/apps/api-gql/internal/delivery/gql/gqlmodel"
	"github.com/twirapp/twir/apps/api-gql/internal/entity"
)

func MapBeRightBackEntityToGQL(e entity.BeRightBackOverlay) gqlmodel.BeRightBackOverlay {
	late := MapBeRightBackLateEntityToGQL(e.Settings.Late)
	return gqlmodel.BeRightBackOverlay{
		ID:              e.ID,
		Text:            e.Settings.Text,
		Late:            &late,
		BackgroundColor: e.Settings.BackgroundColor,
		FontSize:        int(e.Settings.FontSize),
		FontColor:       e.Settings.FontColor,
		FontFamily:      e.Settings.FontFamily,
		CreatedAt:       e.CreatedAt,
		UpdatedAt:       e.UpdatedAt,
		ChannelID:       e.ChannelID,
	}
}

func MapBeRightBackLateEntityToGQL(e entity.BeRightBackOverlayLateSettings) gqlmodel.BeRightBackLate {
	return gqlmodel.BeRightBackLate{
		Enabled:        e.Enabled,
		Text:           e.Text,
		DisplayBrbTime: e.DisplayBrbTime,
	}
}

File: web/layers/dashboard/utils/api/overlays-be-right-back.ts
import { useQuery } from '@urql/vue'
import { createGlobalState } from '@vueuse/core'

import type { OverlaysBeRightBackQuery } from '~/gql/graphql.js'

import { useMutation } from '~/composables/use-mutation'
import { graphql } from '~/gql/gql.js'

export type BeRightBackOverlay = Omit<
	OverlaysBeRightBackQuery['overlaysBeRightBack'],
	'__typename' | 'channel'
>

const invalidationKey = 'BeRightBackOverlayInvalidateKey'

export const useBeRightBackOverlayApi = createGlobalState(() => {
	const useQueryBeRightBack = () =>
		useQuery({
			variables: {},
			context: { additionalTypenames: [invalidationKey] },
			query: graphql(`
				query OverlaysBeRightBack {
					overlaysBeRightBack {
						id
						text
						late {
							enabled
							text
							displayBrbTime
						}
						backgroundColor
						fontSize
						fontColor
						fontFamily
						createdAt
						updatedAt
						channelId
					}
				}
			`),
		})

	const useMutationUpdateBeRightBack = () =>
		useMutation(
			graphql(`
				mutation OverlaysBeRightBackUpdate($input: BeRightBackUpdateInput!) {
					overlaysBeRightBackUpdate(input: $input) {
						id
						text
						late {
							enabled
							text
							displayBrbTime
						}
						backgroundColor
						fontSize
						fontColor
						fontFamily
						createdAt
						updatedAt
						channelId
					}
				}
			`),
			[invalidationKey]
		)

	return {
		useQueryBeRightBack,
		useMutationUpdateBeRightBack,
	}
})
